product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.2-Route-messages-based-on-the-given-regular-expression/5.1.2.1-Using-switch-mediator-on-given-regular-expressions/readme.md

```diff
@@ -19,34 +19,60 @@ Given below is the syntax of how regular expression can be add to the switch med
 
 ```
 
-## Sample use-case
+## Sample use case
+Let's take an example of a client that could send stock quote requests , and receive and display the price of stock quote.
+The client add the name of the stock name which he needs to ge the price of the requested stock.
+So now the request payload includes the above details. When the user sends the request,
+the message is routed to the relevant endpoint by using Switch/Filter mediator.
+In this example, the client can configure from which element or the attribute that the mediator should read and call the back-end service.
 
+Sample request payload
 
-## Pre-requisites
+```
+<m:GetStockPrice xmlns:m="http://www.example.org/stock">
+   <m:StockName>IBM</m:StockName>
+</m:GetStockPrice>
 
+```
 
-## Development guidelines
+Following is a sample synapse configuration in EI.
 
+```
+   <resource methods="POST">
+      <inSequence>
+         <property xmlns:m="http://www.example.org/stock" name="StockName" expression="//m:GetStockPrice/m:StockName" scope="default" type="STRING"/>
+         <switch source="get-property('StockName')">
+            <case regex="IBM">
+               <call>
+                  <endpoint key="http://ei-backend.scenarios.wso2.org:9090/eiTests/XMLEndpoint"/>
+               </call>
+            </case>
+            <default/>
+         </switch>
+         <respond description="PASS"/>
```

Comment:
Do we need to add a description? Even if the payload doesn't match the regex, the message flow will hit here.